### DISCUSSION TITLE
Redirect URL in Github Authorization Not Working

### DISCUSSION THREAD

**Discussion by AurelliaChristie** [supabase/supabase#12945] [Questions] [answered]
# Bug report

## Describe the bug

I'm currently working on a Next.JS app with a `Login with Github` button connected to the Supabase OAuth Github provider. I want the user to be redirected to where the login button is after the authorization is done (currently the login button is only on the `localhost:3000/nominate` page). I've tried several ways I found in the previous issues opened (e.g. [#1895](https://github.com/supabase/supabase/issues/1895)) , but I constantly got redirected to the home page after the Github authorization. Can you please help me on getting things right? Thank you!

## To Reproduce

Here are several ways I've tried:

1. Add the `redirectTo` parameter in the login function as follows:

```
async function signInWithGitHub() {
    const { data, error } = await supabase.auth.signInWithOAuth({
        provider: 'github'
    }, {
        redirectTo: 'http://localhost:3000/nominate/'
    })
        
    if (error) {
        console.log(error);
    }
}
``` 

I also tried to remove the trailing slash of the URL but still not working. Also tried `${window.location.origin}/nominate` but not working as well.

2. Add the following URLs (one per time) to the Redirect URL section on the `Authentication/URL Configuration` page of the Supabase dashboard:

- http://localhost:3000/
- http://localhost:3000/*/
- http://localhost:3000/nominate/
- http://127.0.0.1:3000/
- http://127.0.0.1:3000/*/
- http://127.0.0.1:3000/nominate/
- All the URLs above without the trailing slash 

## Expected behavior

Redirect back to the `http://localhost:3000/nominate` page after the GitHub Authorization.

## System information

- OS: Windows
- Browser: Chrome, Edge
- Version of supabase-js: 2.10.0
- Version of Node.js: 18.14.2

**Comment by GaryAustin1** [comment, marked as answer]
You really should seek help first in Supabase Github or Discussion here before reporting a bug.
Your call does not seem properly formatted (needs `options: {}`) ...
```
const { data, error } = await supabase.auth.signInWithOAuth({
  provider: 'github'
  options: {
    redirectTo: 'https://example.com/welcome'
  }
})
```
Moving to discussions for now.

**Reply by AurelliaChristie** [reply]
Thank you for the answer. And sorry for reporting the bug before seeking help in the GitHub Discussion. Here's what I do to redirect to the last page:

```
  const { data, error } = await supabase.auth.signInWithOAuth({
      provider: 'github',
      options: {
          // redirect to their last page
          redirectTo: window.location.href
      }
  });
```
Thank you once again!

**Reply by DiiiaZoTe** [reply]
I am having the same issue with google provider.
```js
const { data, error } = await supabase.auth.signInWithOAuth({
  provider: "google",
  options: {
    redirectTo: 'http://localhost:3000/account'
  },
});
 ```
 
 I checked the data.url that I get back and this is what I see:
 `htpp://locahost:300 ... =http%3a%2f%2flocalhost%3a3000%2faccount`
 
 Instead of redirecting to the account route, it just refreshes the page...
 Any idea what's going on?

**Comment by lsgrep** [comment]
I am also having this issue.

**Reply by GaryAustin1** [reply]
There appear to be at least 2 different issues above.   And redirect not working can be 1/2 a dozen different things.
Why not start a new discussion and give some details.  Not sure how "I am having this issue" works on this post as the OP got a solution and the 2nd poster seemed to have something else going on.
